subsys/bluetooth/mesh/sensor_srv.c

```diff
@@ -310,12 +333,44 @@ static int handle_series_get(struct bt_mesh_model *model, struct bt_mesh_msg_ctx
 	bt_mesh_model_msg_init(&rsp, BT_MESH_SENSOR_OP_SERIES_STATUS);
 	net_buf_simple_add_le16(&rsp, id);
 
-	if (!sensor) {
+	if (!sensor || !sensor->series.get) {
+		BT_WARN("No series support in 0x%04x", sensor->type->id);
+		goto respond;
+	}
+
+	if (sensor->type->channel_count < 3) {
+		uint16_t start = 0;
+		uint16_t end = sensor->series.column_count - 1;
+
+		if (buf->len == sizeof(uint16_t) * 2) {
+			start = net_buf_simple_pull_le16(buf);
+			end = net_buf_simple_pull_le16(buf);
+			if (start >= sensor->series.column_count ||
+			    end >= sensor->series.column_count ||
+			    start < end) {
+				BT_WARN("Invalid range");
+				goto respond;
+			}
+		} else if (buf->len != 0) {
+			/* invalid length */
+			BT_WARN("Invalid length (%u)", buf->len);
+			return -EMSGSIZE;
+		}
+
+		for (uint32_t i = start; i <= end; i++) {
+			int err = sensor_column_value_encode(&rsp, srv, sensor,
+							     ctx, i);
+
+			if (err) {
+				BT_WARN("Column encode failed");
+				return err;
+			}
+		}
```

Comment:
4.3.1.2.8 from spec rev:
![image](https://user-images.githubusercontent.com/35562897/169851047-ce1d4224-1b40-461b-aa0b-52d92190d7de.png)
Does this implementation cover this? Will a get message without specifying start/end boundaries ensure that a status response contains N number of columns that fits within a payload if the total amount of available columns are greater than what fits into a payload?